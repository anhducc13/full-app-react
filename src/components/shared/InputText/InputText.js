import React from 'react';
import TextField from '@material-ui/core/TextField';

export const InputText = (props) => {
  const { name, label, textField, type, onChange} = props;

  return (
    <TextField
      variant="outlined"
      margin="normal"
      required
      fullWidth
      id={name}
      label={label}
      name={name}
      type={type || "text"}
      value={textField.value}
      error={textField.error !== ''}
      helperText={textField.valid ? '' : textField.error}
      onChange={onChange}
    />
  )
}
