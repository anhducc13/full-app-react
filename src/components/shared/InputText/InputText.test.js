import React from 'react';
import { shallow, mount } from 'enzyme';
import { InputText } from './InputText'

const defaultProps = {
  name: "", 
  label: "", 
  textField: {
    value: "",
    error: "",
    valid: false
  }, 
  type: "text", 
  onChange: jest.fn()
}

describe('Input Text', () => {
  it('should render correctly', () => {
    const component = shallow(<InputText {...defaultProps}/>);
    expect(component).toMatchSnapshot();
  });

  it('function onChange have been called when change input', () => {
    const component = mount(<InputText {...defaultProps}/>);
    component.find('input').simulate('change');
    expect(defaultProps.onChange).toHaveBeenCalled();
    component.unmount();
  })
});

