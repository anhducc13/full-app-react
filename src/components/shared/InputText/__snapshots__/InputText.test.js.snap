// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Text should render correctly 1`] = `
<WithStyles(ForwardRef(TextField))
  error={false}
  fullWidth={true}
  helperText=""
  id=""
  label=""
  margin="normal"
  name=""
  onChange={[MockFunction]}
  required={true}
  type="text"
  value=""
  variant="outlined"
/>
`;
